web-api-shared/src/elastic.rs

```diff
@@ -323,14 +482,15 @@ impl Client {
 
         let status = response.status();
         if status == StatusCode::NOT_FOUND {
-            Ok(None)
+            Err(Error::ResourceNotFound(path))
         } else if !status.is_success() {
             let url = response.url().clone();
             let body = response.bytes().await?;
             let error = String::from_utf8_lossy(&body).into_owned();
             Err(Error::Status { status, url, error })
         } else {
-            Ok(Some(response.json().await?))
+            let body = response.bytes().await?;
+            Ok(serde_json::from_slice(&body)?)
```

Comment:
This was a bug as it would report (de-)serialization errors as `Error::Transport` which is bad as we don't want to retry if there is an (de-)serialization error (through we do want to retry if awaiting the body fails, like we still do).